docs/user/ProofCaching.md

```diff
@@ -0,0 +1,96 @@
+# Proof Caching
+
+-- *Author: Arne Keller*, June 2023 valid for KeY-2.12.0
+
+!!! abstract
+
+    This note describes the new *Proof Caching* functionality in KeY 2.12.
+    It allows reusing parts of earlier proofs when performing another proof.
+   
+
+## Overview
+
+The *Proof Caching* functionality allows reusing previous proofs when creating a new proof.
+More precisely, it allows you to close a branch of the proof by "reference" to an already closed branch in another proof.
+These conditions have to be met:
+
+![tree with cached goals](./ProofCachingTree.png)
+
+## Required conditions for caching to work
+
+To close a branch with sequent S in proof P using another branch with sequent S' in proof P', the following conditions have to bet:
+
+1. Choice settings (handling of integer overflows etc.) of P and P' need to be equal
+2. Antecedent(S') ⊆ Antecedent(S) and Succedent(S') ⊆ Succedent(S)
+3. Neither branch uses state merging
+4. S' may not contain query terms
+5. S' may not contain symbolic execution terms
```

Comment:
What are "symbolic excecution terms"? Do you mean "modalities"?

Reply:
Yes, I have corrected the text.